apps/member/views.py

```diff
@@ -77,3 +78,28 @@ def revoke_resource_request(request, pk):
     if getattr(resource_request, 'resourcegrant', None):
         resource_request.resourcegrant.delete()
     return redirect(reverse('member:dashboard'))
+
+
+@login_required(login_url='home')
+def get_member_data(request, pk, resource_name, record_type):
+    # Get the path to the resource class from the settings, based on the resource_name.
+    resource_class_path = settings.RESOURCE_NAME_AND_CLASS_MAPPING.get(resource_name, None)
+    # If there is not a path for the resource_name, raise an error
+    if not resource_class_path:
+        raise Http404
+
+    # Is the record_type is not valid, raise an error
+    if record_type not in settings.VALID_MEMBER_DATA_RECORD_TYPES:
+        raise Http404
+
+    # Does the request.user's Organization have access to this member's resource?
+    resource_grant = get_object_or_404(
+        ResourceGrant.objects.filter(
+            member_id=pk,
+            resource_class_path=resource_class_path,
+            organization__users=request.user
+        )
+    )
+
+    data = resource_grant.resource_class(resource_grant.member).get()
```

Comment:
The `get` function needs to accept something about the resource you're trying to get, in this case it looks like that would be `record_type`